diff -uNr Python-3.9.2/Lib/http/server.py Python-3.9.2.mod/Lib/http/server.py
--- Python-3.9.2/Lib/http/server.py	2021-02-19 14:31:44.000000000 +0200
+++ Python-3.9.2.mod/Lib/http/server.py	2021-03-20 16:08:34.173543081 +0200
@@ -1165,10 +1165,6 @@
                 return
             # Child
             try:
-                try:
-                    os.setuid(nobody)
-                except OSError:
-                    pass
                 os.dup2(self.rfile.fileno(), 0)
                 os.dup2(self.wfile.fileno(), 1)
                 os.execve(scriptfile, args, env)
